cmake/linker/lld/linker_libraries.cmake

```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2024 Nordic Semiconductor
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set_linker_property(NO_CREATE TARGET linker PROPERTY c_library "-lc")
+# Default per standard, will be populated by clang/target.cmake based on clang output.
+set_linker_property(NO_CREATE TARGET linker PROPERTY rt_library "")
+set_linker_property(TARGET linker PROPERTY c++_library "-lc++;-lc++abi")
```

Comment:
should this mention `math_library` somehow? Either noting that it's not needed or setting to the empty string?

Reply:
II don't think so.
The lib names are not fixed in that sense, a linker implementation can define extra libraries to always link in, as described here:
https://github.com/zephyrproject-rtos/zephyr/blob/3b8e621e9cd6b8382ccdaad4bd757012dafe3d59/cmake/linker/linker_libraries_template.cmake#L9-L10

I will be working on a docset to describe the whole toolchain infrastructure and how to implement support for a custom one.
And there the math lib could be a good candidate to show.